### PR TITLE
qdrant: init at 0.9.1

### PR DESCRIPTION
**Pull request by dit7ya**
###### Description of changes

Qdrant (read: quadrant ) is a vector similarity search engine. It provides a production-ready service with a convenient API to store, search, and manage points - vectors with an additional payload. Qdrant is tailored to extended filtering support. It makes it useful for all sorts of neural-network or semantic-based matching, faceted search, and other applications.

Qdrant is written in Rust crab, which makes it fast and reliable even under high load.

With Qdrant, embeddings or neural network encoders can be turned into full-fledged applications for matching, searching, recommending, and much more!

https://github.com/qdrant/qdrant

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin

- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
- and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)

- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by risicle** [comment]
On macos 10.15:

```
running: "/nix/store/md0r2kkc0wg2gk18aaim529krci9kv3w-clang-wrapper-11.1.0/bin/c++" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-arch" "x86_64" "-I" "rocksdb/include/" "-I" "rocksdb/" "-I" "rocksdb/third-party/gtest-1.8.1/fused-src/" "-I" "snappy/" "-I" "." "-Wall" "-Wextra" "-std=c++17" "-Wsign-compare" "-Wsh
  cargo:warning=In file included from rocksdb/cache/fast_lru_cache.cc:10:
  cargo:warning=rocksdb/cache/fast_lru_cache.h:165:3: error: aligned deallocation function of type 'void (void *, std::align_val_t) noexcept' is only available on macOS 10.14 or newer
  cargo:warning=  ~LRUCacheShard() override = default;
  cargo:warning=  ^
  cargo:warning=rocksdb/cache/fast_lru_cache.cc:439:18: note: in defaulted destructor for 'rocksdb::fast_lru_cache::LRUCacheShard' first required here
  cargo:warning=      shards_[i].~LRUCacheShard();
  cargo:warning=                 ^
  cargo:warning=rocksdb/cache/fast_lru_cache.h:165:3: note: if you supply your own aligned allocation functions, use -faligned-allocation to silence this diagnostic
  cargo:warning=  ~LRUCacheShard() override = default;
  cargo:warning=  ^
  cargo:warning=1 error generated.
  exit status: 1
```

This will be because we still use the macos 10.12 (?) sdk on x86_64.

**Comment by risicle** [comment]
Hmmm https://github.com/NixOS/nixpkgs/pull/179399/commits/b743bab34c3ea756a2170d3717c68f78073854f8

**Comment by risicle** [comment]
```
NIX_CFLAGS_COMPILE = lib.optional stdenv.cc.isClang "-faligned-allocation";
```

indeed solves the issue.

**Comment by risicle** [comment]
WFM - could you squash the commits please?

**Comment by dit7ya** [comment]
Thanks @risicle :)

File: pkgs/servers/search/qdrant/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+, stdenv
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "qdrant";
+  version = "0.9.1";
+
+  src = fetchFromGitHub {
+    owner = "qdrant";
+    repo = "qdrant";
+    rev = "refs/tags/v${version}"; # just v${version} fails to unpack as there is a branch of the same name
```
Comment (SuperSandro2000) [inline pkgs/servers/search/qdrant/default.nix:14]:
```suggestion
    rev = "refs/tags/v${version}";
```

File: pkgs/servers/search/qdrant/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, protobuf
+, stdenv
+}:
+rustPlatform.buildRustPackage rec {
```
Comment (SuperSandro2000) [inline pkgs/servers/search/qdrant/default.nix:7]:
```suggestion
}:

rustPlatform.buildRustPackage rec {
```

**Pull request merged**